datashader/transfer_functions/__init__.py

```diff
@@ -574,20 +575,106 @@ def spread(img, px=1, shape='circle', how='over', mask=None, name=None):
         raise ValueError("mask must be a square 2 dimensional ndarray with "
                          "odd dimensions.")
         mask = mask if mask.dtype == 'bool' else mask.astype('bool')
-    kernel = _build_spread_kernel(how)
+
     w = mask.shape[0]
     extra = w // 2
-    M, N = img.shape
-    buf = np.zeros((M + 2*extra, N + 2*extra), dtype='uint32')
-    kernel(img.data, mask, buf)
-    out = buf[extra:-extra, extra:-extra].copy()
-    return Image(out, dims=img.dims, coords=img.coords, name=name)
+    float_type = img.dtype in [np.float32, np.float64]
+
+    if is_image:
+        kernel = _build_spread_kernel(how, is_image)
+    elif float_type:
+        kernel = _build_float_kernel(how, w)
+    else:
+        kernel = _build_int_kernel(how, w, img.dtype == np.uint32)
+
+    if is_image:
+        M, N = img.shape
+        buf = np.zeros((M + 2*extra, N + 2*extra),
+                       dtype='uint32' if is_image else img.dtype)
```

Comment:
`is_image` is always true in this branch, so it will never use `else img.dtype`. But is there some reason to hardcode it here to `uint32`?  Won't `img.dtype` already be `uint32`?  I think it's better to use `img.dtype` consistently instead of hardcoding it, but maybe you had some reason.

Reply:
I think I agree: that code was left over from an earlier codepath that is no longer in used. I'll using `img.dtype`.

Reply:
Addressed in 9facd5c